containers/Dockerfile.buildenv

```diff
@@ -225,9 +225,9 @@ RUN git clone --single-branch --branch v6.10.2 --depth 1 \
     && cd /work/charm_6_10_2 \
     && git checkout v6.10.2 \
     && ./build LIBS multicore-linux-x86_64 gcc \
-      ${PARALLEL_MAKE_ARG} -g -O0 --build-shared \
+      ${PARALLEL_MAKE_ARG} -g -O2 --build-shared \
```

Comment:
The comment says "we do not build with debug symbols", so should this `-g` be deleted?

Reply:
Ah, yea. This is a back and forth thing. If we need to debug something with `pup` it is potentially useful to have the debug symbols. I'm not sure which is the right thing to do, to be honest. Basically it's:
1. smaller container by not have debug symbols (not sure how much smaller)
2. have debug symbols to make debugging Charm++-related issues easier for folks using the container

Reply:
I think we should make the container debug-friendly, since it's our dev and debug environment. That's to a reasonable extent of course, so if debug symbols for charm were unreasonably large we could consider omitting them, but that's probably not the case. If we end up running production simulations in the container we should make a production container anyway, which would omit a lot of debug/CI things like the older compiler versions.

Reply:
@nilsdeppe do you want to make another change to the Dockerfile or should we merge as-is, potentially making further changes in follow-up PRs?

Reply:
Rebased and squashed in updated comment :)